kong/plugins/datadog/handler.lua

```diff
@@ -106,25 +106,23 @@ local function log(premature, conf, message)
     request_count    = 1,
   }
 
-  local logger, err = statsd_logger:new(conf)
-  if err then
-    ngx_log(NGX_ERR, "failed to create Statsd logger: ", err)
-    return
-  end
-
   for _, metric_config in pairs(conf.metrics) do
-    local metric = metrics[metric_config.name]
+    local metric_func = metrics[metric_config.name]
```

Comment:
since `metrics` is a table of functions it's implied that `metric` is a function. Please revert the name to `metric`.